### DISCUSSION TITLE
Plugin and events

### DISCUSSION THREAD

**Discussion by cosmy81** [saleor/saleor#9145] [General]
Hi all. 
I can't understand how to trigger a plugin. 
I've written a a plugin to add a contact to mailchimp but I don't know how to activate it and how to pass arguments to it. 
In my idea I'd like the plugin to listen to order placed event and to use the order parameters, but I don't know how to do this, in the documentation is not clear to me.
Anyone can help me please?
At the end I can release the code for mailchimp

**Comment by krzysztofwolski** [comment]
I would choose the "app way" and create a separate app for such integration https://docs.saleor.io/docs/3.0/developer/extending/apps/key-concepts .
The app would listen for the `ORDER_CREATED` webhook and send the data to Mailchimp. 
It would be possible to create such integration using low-code solutions like n8n.

**Comment by cosmy81** [comment]
Thank you for your reply.
If I've understood, the app way is to create a separate app or link a third party service in order to receive data and do something with them. 
Ok, it's cool and usefull in some cases (separating core and integrations and use them also with other projects) but not always.
I prefer not to build another app or endpoint and also having the capability to use core models and methods  is important in order to manipulate data and db. 
In the case of mailchimp I don't want to create a service to manage it. It's a simple asyncrounous call to their API.

So my idea is to do this with a celery task in a plugin (the other fastest way is to modify the core but i want to do the thing in the right way), but i can understand the basics: how to say to saleor to call www.mysaleorcoreaddress.io/plugins/my.plugin/webhook/add_contact after an order is created.

**Comment by cosmy81** [comment]
OK, I've found how to manage plugins. 
In base_plugin.py there is the plugin interface with, so I only need to overwrite base methods that are called by saleor core. 
So plugin can act with that base methods.
Everything not managed with that methods need integration in core.
It wasn't so clear to me. Now I'll try to integrate some plugin.
